AsyncDisplayKit/Layout/ASDimension.h

```diff
@@ -12,56 +12,183 @@
+extern ASDimension const ASDimensionAuto;
 
 ASDISPLAYNODE_EXTERN_C_BEGIN
 NS_ASSUME_NONNULL_BEGIN
 
-#pragma mark - ASRelativeDimension
+#pragma mark - ASDimension
 
-extern ASRelativeDimension ASRelativeDimensionMake(ASRelativeDimensionType type, CGFloat value);
+/**
+ * Returns a dimension with the specified type and value.
+ */
+ASOVERLOADABLE extern ASDimension ASDimensionMake(ASDimensionType type, CGFloat value);
 
-extern ASRelativeDimension ASRelativeDimensionMakeWithPoints(CGFloat points);
+/**
+ * Returns a dimension with the specified points value.
+ */
+ASOVERLOADABLE extern ASDimension ASDimensionMake(CGFloat points);
 
-extern ASRelativeDimension ASRelativeDimensionMakeWithFraction(CGFloat fraction);
+/**
+ * Returns a dimension by parsing the specified dimension string.
+ * Examples: ASDimensionMake(@"0.5%") = ASDimensionMake(ASDimensionTypeFraction, 0.5)
+ *           ASDimensionMake(@"0.5pt") = ASDimensionMake(ASDimensionTypePoints, 0.5)
+ */
+ASOVERLOADABLE extern ASDimension ASDimensionMake(NSString *dimension);
 
-extern ASRelativeDimension ASRelativeDimensionCopy(ASRelativeDimension aDimension);
+/**
+ * Returns a dimension with the specified points value.
+ */
+extern ASDimension ASDimensionMakeWithPoints(CGFloat points);
+
+/**
+ * Returns a dimension with the specified fraction value.
+ */
+extern ASDimension ASDimensionMakeWithFraction(CGFloat fraction);
+
+/**
+ * Returns a dimension with the same type and value as the specified dimension.
+ */
+extern ASDimension ASDimensionCopy(ASDimension aDimension);
+
+/**
+ * Returns whether two dimensions are equal.
+ */
+extern BOOL ASDimensionEqualToDimension(ASDimension lhs, ASDimension rhs);
 
-extern BOOL ASRelativeDimensionEqualToRelativeDimension(ASRelativeDimension lhs, ASRelativeDimension rhs);
+/**
+ * Returns a NSString representation of a dimension.
+ */
+extern NSString *NSStringFromASDimension(ASDimension dimension);
 
-extern NSString *NSStringFromASRelativeDimension(ASRelativeDimension dimension);
+/**
+ * Resolve this dimension to a parent size.
+ */
+extern CGFloat ASDimensionResolve(ASDimension dimension, CGFloat autoSize, CGFloat parent);
 
-extern CGFloat ASRelativeDimensionResolve(ASRelativeDimension dimension, CGFloat parent);
+#define ASD(...) ASDimensionMake(__VA_ARGS__)
+
+@interface NSNumber (ASRelativeDimension)
+@property (nonatomic, readonly) ASDimension as_points;
+@property (nonatomic, readonly) ASDimension as_fraction;
+@end
+
+#pragma mark - ASSize
+
+/**
+ * Returns an ASSize with default values.
+ */
+extern ASSize ASSizeMake();
+
+/**
+ * Returns an ASSize with the specified CGSize values as width and height.
+ */
+extern ASSize ASSizeMakeFromCGSize(CGSize size);
+
+/**
+ * Returns whether two sizes are equal.
+ */
+extern BOOL ASSizeEqualToSize(ASSize lhs, ASSize rhs);
+
+/**
+ * Returns a string formatted to contain the data from an ASSize.
+ */
+extern NSString *NSStringFromASSize(ASSize size);
+
+/**
+ * Resolve this size to a parent size.
```

Comment:
This function needs a lot more explanation. It's a key computation of a node's constrained size and should be well understood by anyone who is reading this code.